src/client.ts

```diff
@@ -7,16 +7,28 @@ import { version } from "../package.json";
 import { ManagementController } from "./facades/management";
 
 export class EdcConnectorClient {
-  readonly management: ManagementController;
-  readonly observability: ObservabilityController;
-  readonly public: PublicController;
+
+  apiToken: string | undefined;
+  addresses: Addresses = {};
+  inner = new Inner();
 
   constructor() {
-    const inner = new Inner();
 
-    this.management = new ManagementController(inner);
-    this.observability = new ObservabilityController(inner);
-    this.public = new PublicController(inner);
+  }
+
+  get management() {
+    const context = new EdcConnectorClientContext(this.apiToken, this.addresses);
+    return new ManagementController(this.inner, context);
+  }
+
+  get observability() {
+    const context = new EdcConnectorClientContext(this.apiToken, this.addresses);
+    return new ObservabilityController(this.inner, context);
+  }
+
+  get public() {
+    const context = new EdcConnectorClientContext(this.apiToken, this.addresses);
+    return new PublicController(this.inner, context);
   }
```

Comment:
to be honest I think that the context has some design flaws like, the apiToken and all the URLs were mandatory (???), to build one there's a method on the client... smells not that good.
What could be done (I will provide in a PR) would be to overload the getters on the context to throw an exception if a field that hasn't been set is requested, so it would be easier to be debugged, but I really think these checks should be lazy, otherwise we'd need to check everything everywhere (duplicated logic).